src/objects/zcl_abapgit_object_otgr.clas.abap

```diff
@@ -103,30 +105,49 @@ CLASS ZCL_ABAPGIT_OBJECT_OTGR IMPLEMENTATION.
 
 
   METHOD zif_abapgit_object~deserialize.
-    DATA: ls_otgr        TYPE ty_otgr,
-          lo_otgr        TYPE REF TO cl_cls_object_type_group,
-          lx_pak_error   TYPE REF TO cx_root,
-          lv_text        TYPE string,
-          lv_masterlang  TYPE sylangu.
+    DATA: ls_otgr       TYPE ty_otgr,
+          lo_otgr       TYPE REF TO cl_cls_object_type_group,
+          lx_pak_error  TYPE REF TO cx_root,
+          lv_text       TYPE string,
+          lv_masterlang TYPE sylangu,
+          lo_parents    TYPE REF TO data.
 
     FIELD-SYMBOLS: <ls_groupt>  LIKE LINE OF ls_otgr-texts,
-                   <ls_element> LIKE LINE OF ls_otgr-elements.
-*                   <ls_parent>  LIKE LINE OF ls_otgr-parents.
+                   <ls_element> LIKE LINE OF ls_otgr-elements,
+                   <lv_field>   TYPE any,
+                   <ls_parent>  TYPE any,
+                   <lt_parents> TYPE ANY TABLE.
 
     io_xml->read( EXPORTING iv_name = 'OTGR'
-                  CHANGING cg_data = ls_otgr ).
+                  CHANGING  cg_data = ls_otgr ).
 
     LOOP AT ls_otgr-texts ASSIGNING <ls_groupt>.
       <ls_groupt>-activation_state = cl_pak_wb_domains=>co_activation_state-inactive.
       " Removed in the method serialize.
       <ls_groupt>-name = ms_item-obj_name.
     ENDLOOP.
 
-*    LOOP AT ls_otgr-parents ASSIGNING <ls_parent>.
-*      <ls_parent>-activation_state = cl_pak_wb_domains=>co_activation_state-inactive.
-*      " Removed in the method serialize.
-*      <ls_parent>-obj_type_group = ms_item-obj_name.
-*    ENDLOOP.
+    " Parents (cls_tygr_parent) does not exist in lower releases
+    TRY.
+        CREATE DATA lo_parents TYPE TABLE OF ('CLS_TYGR_PARENT').
+        ASSIGN lo_parents->* TO <lt_parents>.
+
+        io_xml->read( EXPORTING iv_name = 'PARENTS'
+                      CHANGING  cg_data = <lt_parents>  ).
+
+        LOOP AT <lt_parents> ASSIGNING <ls_parent>.
+          ASSIGN COMPONENT 'ACTIVATION_STATE' OF STRUCTURE <ls_parent> TO <lv_field>.
+          IF sy-subrc = 0.
+            <lv_field> = cl_pak_wb_domains=>co_activation_state-inactive.
+          ENDIF.
+          ASSIGN COMPONENT 'OBJ_TYPE_GROUP' OF STRUCTURE <ls_parent> TO <lv_field>.
+          IF sy-subrc = 0.
+            " Removed in the method serialize.
+            <lv_field> = ms_item-obj_name.
+          ENDIF.
+        ENDLOOP.
+      CATCH cx_root.
```

Comment:
Is this the best what we can do? Catching the top level exception and not spitting a word looks like an anti pattern, especially in such a huge code block.

Where do you expect the first occurrence of exception? In `CREATE DATA` or `io_xml->read`?

Reply:
If anything fails in this block, it means the feature is not supported by this release. I will try to reduce try catch and make it more specific.

Reply:
That's the devil - what if the constructor of `CLS_TYGR_PARENT` start raising a new exception in the case of XYZ? Or `io_xml->read()`? We will never find out.

Reply:
fixed